File: spare_parts_shop_flutter/lib/models/dashboard_stats.dart
class DashboardStats {
  final double todaySales;
  final double weeklySales;
  final double monthlySales;
  final int todayBillsCount;
  final int lowStockCount;
  final int totalProducts;

  DashboardStats({
    required this.todaySales,
    required this.weeklySales,
    required this.monthlySales,
    required this.todayBillsCount,
    required this.lowStockCount,
    this.totalProducts = 0,
  });

  factory DashboardStats.fromJson(Map<String, dynamic> json) {
    return DashboardStats(
      todaySales: (json['todaySales'] as num).toDouble(),
      weeklySales: (json['weeklySales'] as num).toDouble(),
      monthlySales: (json['monthlySales'] as num).toDouble(),
      todayBillsCount: json['todayBillsCount'],
      lowStockCount: json['lowStockCount'],
      totalProducts: json['totalProducts'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'todaySales': todaySales,
      'weeklySales': weeklySales,
      'monthlySales': monthlySales,
      'todayBillsCount': todayBillsCount,
      'lowStockCount': lowStockCount,
      'totalProducts': totalProducts,
    };
  }
}

File: spare-parts-shop/src/main/java/com/spareparts/dto/DashboardStats.java
package com.spareparts.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class DashboardStats {
    private Double todaySales;
    private Double weeklySales;
    private Double monthlySales;
    private Long todayBillsCount;
    private Integer lowStockCount;
    private Long totalProducts = 0L;
    
    // New Admin KPIs
    private Integer outOfStockCount = 0;
    private Integer deadStockCount = 0;
    private Integer fastMovingProductsCount = 0;
    private Double netProfit = 0.0;
    private Double gstCollected = 0.0;
    
    private Integer totalCustomers = 0;
    private Integer newCustomers = 0;
    private Integer activeCustomers = 0;
    private Double customerGrowthPercent = 0.0;
}